packages/ses/src/module-instance.js

```diff
@@ -0,0 +1,314 @@
+        set: undefined,
+        enumerable: true,
+        configurable: false,
+      };
+
+      notifiers[liveExportName] = liveGetNotify.notify;
+    },
+  );
+
+  const notifyStar = update => {
+    update(moduleNS);
+  };
+  notifiers['*'] = notifyStar;
+
+  // The updateRecord must conform to linkageRecord.imports
+  // updateRecord = Map<specifier, importUpdaters>
+  // importUpdaters = Map<importName, [update(newValue)*]>
+  function imports(updateRecord) {
+    // By the time imports is called, the importNS should already be
+    // initialized with module instances that satisfy
+    // linkageRecord.imports.
+    // importNS = Map[_specifier_, { initialize, notifiers }]
+    // notifiers = { _importName_: notify(update(newValue))}
+
+    // export * cannot export default.
+    const candidateAll = create(null);
+    candidateAll.default = false;
+    for (const [specifier, importUpdaters] of updateRecord.entries()) {
+      const instance = importNS.get(specifier);
+      instance.getNamespace(); // bottom up cycle tolerant
+      const { notifiers: modNotifiers } = instance;
+      for (const [importName, updaters] of importUpdaters.entries()) {
+        const notify = modNotifiers[importName];
+        if (!notify) {
+          throw SyntaxError(
+            `The requested module '${specifier}' does not provide an export named '${importName}'`,
+          );
+        }
+        for (const updater of updaters) {
+          notify(updater);
+        }
+      }
+      if (linkageRecord.exportAlls.includes(specifier)) {
+        // Make all these imports candidates.
+        for (const [importName, notify] of entries(modNotifiers)) {
+          if (candidateAll[importName] === undefined) {
+            candidateAll[importName] = notify;
+          } else {
+            // Already a candidate: remove ambiguity.
+            candidateAll[importName] = false;
+          }
+        }
+      }
+    }
+
+    for (const [importName, notify] of entries(candidateAll)) {
+      if (!notifiers[importName] && notify !== false) {
+        notifiers[importName] = notify;
+
+        // exported live binding state
+        let value;
+        notify(v => (value = v));
+        moduleNSProps[importName] = {
+          get() {
+            return value;
+          },
+          set: undefined,
+          enumerable: true,
+          configurable: false,
+        };
+      }
+    }
+
+    // Sort the module namespace as per spec.
+    // The namespace will be wrapped in a namespace proxy which will serve as a
+    // "module namespace exotic object".
+    keys(moduleNSProps)
+      .sort()
+      .forEach(k => defProp(moduleNS, k, moduleNSProps[k]));
+  }
+
+  const { functorSource } = linkageRecord;
+  const realmRec = getCurrentRealmRec();
+  let optFunctor = performEval(
+    realmRec,
+    functorSource,
+    globalObject,
+    {},
+    {
+      localTransforms: [],
+      globalTransforms: [],
+      sloppyGlobalsMode: false,
+    },
+  );
```

Comment:
This has changed from the original.

Reply:
Yes, that would be necessary. The reason for omitting them here is that the functor source is a JavaScript program, not an ESM, and we shouldn’t reveal the internal representation of a module to third-party transforms.